project/src/main/webapp/gmail.js

```diff
@@ -0,0 +1,365 @@
+//         'format': "raw"
+//     });
+//     messageRequest.execute(response => {
+//         //Convert from base64 encoding to text.
+//         const decodedEmail = atob(response.raw.replace(/-/g, '+').replace(/_/g, '/'));
+//         const emailBodyValue = getEmailBody(decodedEmail);
+//         const emailDateValue = getEmailDate(decodedEmail);
+//         const messageID = getMessageID(decodedEmail);
+//         if (isActualEmail(emailBodyValue, emailDateValue)) {
+//             emailObjects[messageID] = { emailDate: emailDateValue, emailBody: emailBodyValue };
+//         }
+//         getEmailObjects();
+//     });
+// }
+
+// //Retrieve messages using hardcoded queries and the signed-in email.
+// function listMessages() {
+//     emailObjects = {};
+//     var getPageOfMessages = function(request, result) {
+//         request.execute(function(resp) {
+//             result = result.concat(resp.messages);
+//             var nextPageToken = resp.nextPageToken;
+//             if (nextPageToken) {
+//                 request = gapi.client.gmail.users.messages.list({
+//                     'userId': 'me',
+//                     'pageToken': nextPageToken,
+//                     'q': getQuery()
+//                 });
+//                 getPageOfMessages(request, result);
+//             } else {
+//                 for (var i = 0; i < result.length; i++) {
+//                     if (result[i] === undefined) {
+//                         console.log("There are no messages from the query");
+//                     }
+//                     else {
+//                         console.log(result.length);
+//                         console.log(result);
+//                         this.getMessage(result[i].id);
+//                     }
+//                 }
+//             }
+//         });
+//     };
+//     var initialRequest = gapi.client.gmail.users.messages.list({
+//         'userId': 'me',
+//         'q': getQuery()
+//     });
+//     getPageOfMessages(initialRequest, []);
+// }
+
+class gmailAPI {
+    question;
+    emailObjects = {};
+
+    constructor(question) {
+        this.question = question;
+        this.emailObjects = {};
+    }
+
+
+    getEmailBody(decodedEmail) {
+        const emailBodyStartIndex = decodedEmail.indexOf("Content-Type: text/plain; charset=\"UTF-8\"");
+        const emailBodyEndIndex = (decodedEmail.substring(emailBodyStartIndex)).indexOf("Content-Type: text/html; charset=\"UTF-8\"")
+        let emailBodyValue = decodedEmail.substring(emailBodyStartIndex, emailBodyStartIndex + emailBodyEndIndex);
+        if (emailBodyValue.indexOf("Forwarded message") != -1) {
+            emailBodyValue = emailBodyValue.substring(0, emailBodyValue.indexOf("Forwarded message"));
+        }
+        const quotedPrintableIndex = emailBodyValue.indexOf("quoted-printable");
+        if (quotedPrintableIndex != -1) {
+            emailBodyValue = emailBodyValue.substring(quotedPrintableIndex + 16);
```

Comment:
It's unclear why this 16 is here, can you put it in a variable with a descriptive name?

Reply:
Just did it!